src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchOperations.java

```diff
@@ -326,36 +326,34 @@
 	/**
 	 * return number of elements found by given query
 	 *
-	 * @param query
-	 * @param clazz
-	 * @return
+	 * @param query the query to execute
+	 * @param clazz the entity clazz containing the index information
+	 * @return count
 	 */
-	<T> long count(CriteriaQuery query, Class<T> clazz);
-
-	/**
-	 * return number of elements found by given query
-	 *
-	 * @param query
-	 * @return
-	 */
-	<T> long count(CriteriaQuery query);
+	default long count(Query query, @Nullable Class<?> clazz) {
```

Comment:
Ideally, we can make the entity class non-nullable (if we manage getting rid of the index coordinates inside of `Query`).